src/mono/mono/metadata/native-library.c

```diff
@@ -1222,3 +1222,32 @@ mono_loader_install_pinvoke_override (PInvokeOverrideFn override_fn)
 {
 	pinvoke_override = override_fn;
 }
+
+// Keep synced with FixupSymbolName from src/tasks/Common/Utils.cs
+char* mono_fixup_symbol_name (char *key) {
+	char* fixedName = malloc(256);
+	int sb_index = 0;
+	int len = (int)strlen (key);
+
+	for (int i = 0; i < len; ++i) {
+		unsigned char b = key[i];
+		if ((b >= '0' && b <= '9') ||
+		    (b >= 'a' && b <= 'z') ||
+			(b >= 'A' && b <= 'Z') ||
+			(b == '_')) {
+			fixedName[sb_index++] = b;
+		}
+		else if (b == '.' || b == '-' || b ==  '+' || b == '<' || b == '>') {
+			fixedName[sb_index++] = '_';
+		}
+		else {
+			// Append the hexadecimal representation of b between underscores
+			sprintf(&fixedName[sb_index], "_%X_", b);
+			sb_index += 4; // Move the index after the appended hexadecimal characters
+		}
```

Comment:
we need to add a check that sb_index is < 252 (to account for the possible 4 hex bytes + 1 null terminate) in the loop so we don't write past the fixedName boundary.

Reply:
and ideally use snprintf to make it clear the code is doing this